extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -202,15 +273,173 @@ export class AmpStory360 extends AMP.BaseElement {
     container.appendChild(this.canvas_);
     this.applyFillContent(container, /* replacedContent */ true);
 
-    return Services.storyStoreServiceForOrNull(this.win).then(
-      (storeService) => {
-        storeService.subscribe(
-          StateProperty.PAGE_SIZE,
-          this.resizeRenderer_.bind(this),
-          false /* callToInitialize */
-        );
+    attr('controls') === 'gyroscope' && this.checkGyroscopePermissions_();
+
+    Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
```

Comment:
You want to make this `buildCallback` asynchronous so `layoutCallback` isn't called until the service is retrieved and available. You can simply add a `return` statement to return the service getter's promise.